### ISSUE TITLE
[Feature/pull request] Add an image placeholder

### ISSUE THREAD

**Issue by rvi** [airbnb/lottie-ios#657]
## Check these before submitting:
- [x] Updated to the latest version of Lottie (2.5.0)
- [x] The issue doesn't involve an Unsupported Feature
- [x] This issue isn't related to another open issue**

## This issue is a:
- [ ] Non-Crashing Bug (Visual or otherwise)
- [ ] Crashing Bug
- [x] Feature Request
- [ ] Regression (Something that once worked, but doesn't work anymore)

## What Platform are you on? 

- [ ] MacOS
- [x] iOS

## Adding an image as a placeholder 

On some old devices, the compositions can take sometime to allocate (specially when we have several on the same screen), I would like to be able to add the first frame image as a placeholder.

Would you be interested in a PR? If not I would do only on my code.

Thanks.

**Comment by buba447** [comment]
Hello! 
We would like to keep this functionality out of the main repo, since it would further diverge parity between ios/android. It should be pretty easy to add an image placeholder to a lottie view without altering the library. I would be interested in learning more about slow composition allocation. Are you using any composition caching? Any idea why the compositions are slow to allocate? Can you share one of these compositions?

**Comment by joshwoods** [comment]
@buba447 how would one go about actually pulling off a placeholder image? I have tried using the `AnimationImageProvider` but that doesn't seem to work. We are looking to add something like this for accessibility reasons. If a user has `isReduceMotionEnabled` enabled, we do not want to play the animation but instead display a static image for the users. When I implement `AnimationImageProvider`, I do not even see the `imageForAsset` function called. I have called `reloadImages` on the animation view and still nothing. Would really love to be able to pull this off for our users!
